draft-ietf-quic-transport.md

```diff
@@ -1000,6 +1000,12 @@ to more quickly identify when a connection becomes unusable.
 Packets that are matched to an existing connection, but for which the endpoint
 cannot remove packet protection, are discarded.
 
+Invalid packets without packet protection, such as Initial, Retry, or Version
+Negotiation, SHOULD be discarded without altering connection state.  An endpoint
```

Comment:
I tend to agree with @marten-seemann. I wonder if we might rather say MAY here (or "can"), because people might wonder what the anticipated behavior is, without the definition of what "invalid" means.

Does it mean a packet with a broken encoding? Or does it mean a frame containing a semantically invalid frame; i.e. ACK containing a PN that was never used?

My understanding is that we do not want to define "invalid", hence the suggestion to use a weaker word.

Reply:
I'm not sure this is generally a good idea. Consider the case where we have a packet with two frames. (1) a perfectly legitimate stream frame (2) a frame that is misformatted but generally parseable and doesn't otherwise require connection termination. Why is the sensible procedure here to (a) keep the connection alive but (b) ignore the stream data?

Reply:
Initials are the only packets on the list with multiple frames, and they are AES-GCM encrypted, so an invalid frame indicates a buggy peer or an injection attack.  In both cases parsing part of the packet seems unwise.

Also, if you only process one frame, do you ACK that packet?

Reply:
@ianswett what's our rule here for the same defective packet if it were received protected?

Reply:
I'm not Ian ... but you can't close a connection on valid AEAD but i'll formatted content unless you have advanced the handshake sufficiently due to injection attacks. There is explicit text for a variety of cases already. The current PR covers the rest.

Reply:
@mikkelfj well, that's exactly what we're discussing. And given that we don't have a threat model for this class of attack, I'm not sure that that's true at all.

Reply:
When I say can't close, I mean you should drop packets. Or close if you already affected state as the PR says - even if injection because you can no longer trust yourself - therefore you need to validate carefully before applying state, especially during handshake. You definitely shouldn't try to recover.

That said, perhaps we need to clarify exactly when we switch from dropping to closing. You need to have established a secure line of communication. Bogus TLS can be injection attacks as well as endpoint sourced nonsense.

Reply:
If it was protected, we'd close the connection.